fe/fe-core/src/main/java/com/starrocks/planner/JDBCScanNode.java

```diff
@@ -144,6 +159,93 @@ private void createJDBCTableFilters() {
         }
     }
 
+    private static final Pattern SESSION_VARIABLE_PATTERN = Pattern.compile(
+            "(@'[^']+'|@\"[^\"]+\"|@[A-Za-z0-9_]+|[A-Za-z0-9_]+)\\s*=\\s*('[^']*'|\"[^\"]*\"|\\d+)(?:,|$)"
+    );
+
+    private void createJDBCTableSessionVariableHints() {
+        String jdbcExternalTableSessionVariables =
+                ConnectContext.get().getSessionVariable().getJdbcExternalTableSessionVariables();
+        if (StringUtils.isEmpty(jdbcExternalTableSessionVariables)) {
+            return;
+        }
+        if (!isMysql()) {
+            throw new UnsupportedOperationException(
+                    String.format("Sending session variable to JDBC external table is only supported for MYSQL protocol")
+            );
+        }
+        boolean isStarRocksTable = isStarRocksTable();
+
+        // Prepare clauses for hints
+        StringBuilder setVarHintClause = new StringBuilder();
+        StringBuilder setUserVariableHintClause = new StringBuilder();
+
+        // Split the input into 'variableName=variableValue' string assignments and iterate over each assignment
+        String[] sessionVariableAssignments =
+                jdbcExternalTableSessionVariables.split(",(?=(?:[^']*'[^']*')*[^']*$)");
+        for (String assignment : sessionVariableAssignments) {
+            validateSessionVariableAssignmentSyntax(assignment);
+            // Check if user defined variable: starts with '@' e.g. @var1
+            if (assignment.startsWith("@")) {
+                if (!isStarRocksTable) {
+                    // StarRocks table supports User Defined variable in query hint but not plain MySQL
+                    throw new UnsupportedOperationException("Sending user defined variables to JDBC external table is " +
+                            "only supported for \"database_type\" = 'starrocks'. " + "Invalid assignment: " + assignment +
+                            ". Please add \"database_type\" = \"starrocks\" to the JDBC resource properties on creation " +
+                            "if you intend to propagate user defined variables to an external StarRocks cluster.");
+                }
+                if (setUserVariableHintClause.length() > 0) {
+                    setUserVariableHintClause.append(", ");
+                }
+                setUserVariableHintClause.append(assignment);
+            } else {
+                if (isStarRocksTable) {
```

Comment:
So here you are going to connect another starrocks clutser via JDBC/MySQL protocol?

Reply:
@dirtysalt Yes that's correct 👍